### PR TITLE
feat: Rewrite suggestions and diagnostics in LG LSP with latest LG APIs

### LINKED ISSUE #2211: Implement LG LSP with latest LG APIs
The latest LG APIs has changed. All LG related funcionals are now managed by LGFile instead of LGResources.
LG LSP should be implemented with these APIs.

### PR DESCRIPTION
**Pull request by cosmicshuai**
closes #2211 
Implement completion and validate functions with latest LG APIs. 
Now templates and diagonsitc are managed under LGFile class.
Indexer for LG editor also has been modified. Replace statichecker with LGParser.

### CODE REVIEW

File: Composer/packages/tools/language-servers/language-generation/package.json [hunk 1]
```diff
@@ -16,8 +16,8 @@
   },
   "dependencies": {
     "@bfc/indexers": "*",
-    "botbuilder-lg": "^4.8.0-preview.106823",
-    "botframework-expressions": "^4.8.0-preview.106476",
+    "botbuilder-lg": "^4.8.0-preview-109324",
+    "adaptive-expressions": "^4.8.0-preview-110700",
```
Comment (Danieladu) [inline Composer/packages/tools/language-servers/language-generation/package.json]: maybe both with version "109324" is more robust?

File: Composer/packages/tools/language-servers/language-generation/src/LGServer.ts [hunk 2]
```diff
@@ -214,12 +215,12 @@ export class LGServer {
...
-        imports.forEach(({ id }) => {
-          const importedContent = importResolver('.', id).content;
-          const importedTemplates = parse(importedContent, '');
-          templates.push(...importedTemplates);
+        const lgImports = LGParser.parseText(content, id, importResolver).references;
+        lgImports.forEach(u => {
+          templates.push(...u.templates);
         });
+
+        console.log(templates);
```
Comment (zhixzhan) [inline Composer/packages/tools/language-servers/language-generation/src/LGServer.ts]: Unexpected console.log

File: Composer/packages/tools/language-servers/language-generation/src/utils.ts [hunk 3]
```diff
@@ -123,19 +115,19 @@ export function textFromTemplate(template: Template): string {
...
     // ignore non-exist references in template body.
     return diagnostic.message.includes('does not have an evaluator') === false;
   });
 }
 
 export function updateTemplate(content: string, name: string, body: string): string {
-  const resource = LGParser.parse(content);
-  const template = resource.templates.find(t => t.name === name);
+  const lgFile = LGParser.parseText(content);
+  const template = lgFile.templates.find(t => t.name === name);
```
Comment (Danieladu) [inline Composer/packages/tools/language-servers/language-generation/src/utils.ts]: maybe 
```
return lgFile.templates.some(t => t.name === name) ?
     lgFile.addTemplate(name, [], body).toString() :
     lgFile.updateTemplate(name, name, template.parameters, body).toString();
```
is more concise?

File: Composer/packages/tools/language-servers/language-generation/src/LGServer.ts [hunk 4]
```diff
@@ -443,13 +444,12 @@ export class LGServer {
     const range = getRangeAtPosition(document, position);
     const wordAtCurRange = document.getText(range);
     const endWithDot = wordAtCurRange.endsWith('.');
-    const lgFile = this.getLGDocument(document)?.index();
+    const lgFile = LGParser.parseText(document.getText(), '', this.getImportResolver(document));
```
Comment (zhixzhan) [inline Composer/packages/tools/language-servers/language-generation/src/LGServer.ts]: Seems no need to change this part, the original `this.getLGDocument(document)?.index();` suppose to handle inline case. if change so, `document.getText()` may return inline body, it's not valid for `parseText`.

File: Composer/packages/tools/language-servers/language-generation/src/LGServer.ts [hunk 5]
```diff
@@ -443,13 +444,12 @@ export class LGServer {
...
     if (!lgFile) {
       return Promise.resolve(null);
     }
-    const { templates } = lgFile;
 
-    const completionTemplateList: CompletionItem[] = templates.map(template => {
+    const completionTemplateList: CompletionItem[] = lgFile.allTemplates.map(template => {
```
Comment (zhixzhan) [inline Composer/packages/tools/language-servers/language-generation/src/LGServer.ts]: we can extend return values of `this.getLGDocument(document)?.index()` to cover allTemplates
```
{
  diagnostics,
  templates,      // file itself's
  allTemplates,  // include imports
}
```

File: Composer/packages/server/assets/projects/ActionsSample/ComposerDialogs/Actions/Actions.dialog [hunk 6]
```diff
@@ -16,7 +16,7 @@
         {
           "$type": "Microsoft.SendActivity",
           "$designer": {
-            "id": "522200"
+            "id": "166131"
           },
           "activity": "${bfdactivity-522200()}"
         },
```
Comment (yeze322) [inline Composer/packages/server/assets/projects/ActionsSample/ComposerDialogs/Actions/Actions.dialog]: Just curious about why many `$designer.id` fields are changed? Are those changes intended?

Comment (yeze322) [reply Composer/packages/server/assets/projects/ActionsSample/ComposerDialogs/Actions/Actions.dialog]: Another question is why the designer id here is not equal to lg template's id?

Comment (cosmicshuai) [reply Composer/packages/server/assets/projects/ActionsSample/ComposerDialogs/Actions/Actions.dialog]: > Another question is why the designer id here is not equal to lg template's id?



> Just curious about why many `$designer.id` fields are changed? Are those changes intended?

No, it's not intended. It just changed after I build it. It should be reverted to the original value. Will fix it soon.

**Pull request merged**